### PR TITLE
Move ballerina build, test, and publishing into a single task

### PR DESCRIPTION
**Pull request by lahirusamith**
## Purpose
$Subject 

issue: https://github.com/ballerina-platform/ballerina-standard-library/issues/1247

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog

### CODE REVIEW

File: mysql-ballerina/build.gradle [hunk 1]
```diff
@@ -433,28 +477,16 @@ updateTomlFile.dependsOn copyStdlibs
...
 build.dependsOn ballerinaBuild
 
-ballerinaPublish.finalizedBy revertTomlFile
-ballerinaPublish.dependsOn initializeVariables
-ballerinaPublish.dependsOn updateTomlFile
-ballerinaPublish.dependsOn ":${packageName}-native:build"
-ballerinaPublish.dependsOn ballerinaBuild
-publish.dependsOn ballerinaPublish
+publishToMavenLocal.dependsOn build
+publish.dependsOn build
```
Comment (BuddhiWathsala) [inline mysql-ballerina/build.gradle:492]:
```suggestion
publish.dependsOn build

```

File: mysql-ballerina/build.gradle [hunk 2]
```diff
@@ -154,87 +158,127 @@ task initializeVariables {
...
-
-        if (graph.hasTask(":${packageName}-ballerina:publish")) {
-            ballerinaPublish.enabled = true
-        } else {
-            ballerinaPublish.enabled = false
-        }
     }
 }
 
 task ballerinaTest {
```
Comment (BuddhiWathsala) [inline mysql-ballerina/build.gradle:190]: We should remove this task.

File: mysql-ballerina/build.gradle [hunk 3]
```diff
@@ -433,28 +477,16 @@ updateTomlFile.dependsOn copyStdlibs
...
-ballerinaTest.finalizedBy stopTestDockerContainer
-ballerinaTest.finalizedBy stopTestSslDockerContainer
-test.dependsOn ballerinaTest
-
-ballerinaBuild.dependsOn test
 ballerinaBuild.dependsOn updateTomlFile
 ballerinaBuild.dependsOn initializeVariables
 ballerinaBuild.dependsOn ":${packageName}-native:build"
+ballerinaTest.dependsOn startTestDockerContainer
+ballerinaTest.dependsOn startTestSslDockerContainer
```
Comment (BuddhiWathsala) [inline mysql-ballerina/build.gradle:484]: Need to remove.

**Review by BuddhiWathsala — CHANGES_REQUESTED** [review]

File: mysql-ballerina/build.gradle (hunk 3 above)
Comment (lahirusamith) [reply mysql-ballerina/build.gradle]: fixed in : https://github.com/ballerina-platform/module-ballerinax-mysql/pull/132/commits/ea0e93daab7f4da8146195ed056ccaf6106ba229

File: mysql-ballerina/build.gradle (hunk 2 above)
Comment (lahirusamith) [reply mysql-ballerina/build.gradle]: removed in https://github.com/ballerina-platform/module-ballerinax-mysql/pull/132/commits/ea0e93daab7f4da8146195ed056ccaf6106ba229

File: mysql-ballerina/build.gradle (hunk 1 above)
Comment (lahirusamith) [reply mysql-ballerina/build.gradle]: fixed in https://github.com/ballerina-platform/module-ballerinax-mysql/pull/132/commits/ea0e93daab7f4da8146195ed056ccaf6106ba229

File: mysql-ballerina/build.gradle [hunk 4]
```diff
@@ -154,135 +158,127 @@ task initializeVariables {
...
-        if (repository == "local") {
-            println("Publishing to the ballerina local repository...")
-            balPush("--repository=local")
-        } else if (repository == "central") {
-            // Publish to central
-            println("Publishing to the ballerina central..")
-            if (project.version.endsWith('-SNAPSHOT') ||
-                    project.version.matches(project.ext.timestampedVersionRegex)) {
+        if (needPublishToCentral) {
+            if (project.version.endsWith('-SNAPSHOT')) {
```
Comment (niveathika) [inline mysql-ballerina/build.gradle:248]: Let's add timestamped regex too to the condition

File: mysql-ballerina/build.gradle [hunk 5]
```diff
@@ -433,28 +429,16 @@ updateTomlFile.dependsOn copyStdlibs
...
-ballerinaBuild.dependsOn test
 ballerinaBuild.dependsOn updateTomlFile
 ballerinaBuild.dependsOn initializeVariables
 ballerinaBuild.dependsOn ":${packageName}-native:build"
 ballerinaBuild.finalizedBy revertTomlFile
 ballerinaBuild.finalizedBy stopTestDockerContainer
 ballerinaBuild.finalizedBy stopTestSslDockerContainer
+test.dependsOn startTestDockerContainer
+test.dependsOn startTestSslDockerContainer
+ballerinaBuild.dependsOn test
```
Comment (BuddhiWathsala) [inline mysql-ballerina/build.gradle:440]: I wonder what is the behavior of:
```
./gradlew clean build -x test
./gradlew clean test
```

**Review by BuddhiWathsala — CHANGES_REQUESTED** [review]

File: mysql-ballerina/build.gradle [hunk 6]
```diff
@@ -154,135 +158,127 @@ task initializeVariables {
...
-            println("Publishing to the ballerina local repository...")
-            balPush("--repository=local")
-        } else if (repository == "central") {
-            // Publish to central
-            println("Publishing to the ballerina central..")
-            if (project.version.endsWith('-SNAPSHOT') ||
-                    project.version.matches(project.ext.timestampedVersionRegex)) {
+        if (needPublishToCentral) {
+            if (project.version.endsWith('-SNAPSHOT')) {
                 throw new InvalidUserDataException("Cannot push SNAPSHOT version to central")
```
Comment (niveathika) [inline mysql-ballerina/build.gradle:249]: Change it to return as well

Comment (lahirusamith) [reply mysql-ballerina/build.gradle]: changed in https://github.com/ballerina-platform/module-ballerinax-mysql/pull/132/commits/6a30a525281f84a9aa0f31a7def078a1e0e5141c

File: mysql-ballerina/build.gradle (hunk 5 above)
Comment (lahirusamith) [reply mysql-ballerina/build.gradle]: the command `./gradlew clean build -x test` is to build the project but ignore test cases. this is useful to check any compilation issues in the code without running test cases. on the other hand `./gradlew clean test` is useful when writing and testing test cases without any code changes.

**Review by BuddhiWathsala — APPROVED** [review]
LGTM

**Pull request merged**